### PR TITLE
Datalake TF: Added umccr_datalake warehouse stack

### PR DESCRIPTION
**Pull request by victorskl**
* Initial setup with multiqc Glue cralwers for Parquet data lake

### CODE REVIEW

File: terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        "s3:PutObject"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${bucket}/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "lakeformation:*"
```
Comment (pdiakumis) [inline terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json:17]: Nice, I assume this policy would've helped the other day when I had to try all sorts of lake formation permission combinations to get through these errors: https://aws.amazon.com/premiumsupport/knowledge-center/glue-insufficient-lakeformation-permissions/

**Review by pdiakumis — APPROVED** [review]

**Review by reisingerf — APPROVED** [review]
Sweet!

File: terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json (hunk 1 above)
Comment (victorskl) [reply terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json:17]: Good pointer to the article, Pete. TBH; I just copy in this policy from the Glue role that auto setup through Glue Console. Let me read up the article and, see if I can fine tune a bit. Looks like we could/should tune a bit here, yes.

**Review by brainstorm — APPROVED** [review]
Cool! So nice to see this rolling ;)

File: terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json (hunk 1 above)
Comment (victorskl) [reply terraform/stacks/umccr_datalake/policies/datalake_role_inline_policy.json]: Right! Done a bit of reading. And removed LakeFormation permission with https://github.com/umccr/infrastructure/commit/d7bf0859418e9e506d3ef578d2ba84cca4bc2424

* Turn out we do not need LakeFormation permission just yet as we can just use Glue default IAM-based permission.
* We can upgrade to LakeFormation access control model in future when it sees fit for more complex access control use cases.
* For that, see
https://docs.aws.amazon.com/lake-formation/latest/dg/upgrade-glue-lake-formation.html

**Comment by victorskl** [comment]
Thanks all, merging...

**Pull request merged**